webapp/src/main/java/com/box/l10n/mojito/service/delta/DeltaService.java

```diff
@@ -0,0 +1,192 @@
+package com.box.l10n.mojito.service.delta;
+
+import com.box.l10n.mojito.entity.BaseEntity;
+import com.box.l10n.mojito.entity.Locale;
+import com.box.l10n.mojito.entity.PullRun;
+import com.box.l10n.mojito.entity.PushRun;
+import com.box.l10n.mojito.entity.Repository;
+import com.box.l10n.mojito.entity.RepositoryLocale;
+import com.box.l10n.mojito.entity.TMTextUnitVariant;
+import com.box.l10n.mojito.service.commit.CommitService;
+import com.box.l10n.mojito.service.delta.dtos.DeltaLocaleDataDTO;
+import com.box.l10n.mojito.service.delta.dtos.DeltaMetadataDTO;
+import com.box.l10n.mojito.service.delta.dtos.DeltaResponseDTO;
+import com.box.l10n.mojito.service.delta.dtos.DeltaTranslationDTO;
+import com.box.l10n.mojito.service.repository.RepositoryService;
+import com.box.l10n.mojito.service.tm.TMTextUnitVariantRepository;
+import com.box.l10n.mojito.service.tm.TextUnitVariantDelta;
+import org.joda.time.DateTime;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Service that enables delta generation.
+ *
+ * @author garion
+ */
+@Service
+public class DeltaService {
+    final
```

Comment:
in most places we don't use final, not saying it is better but maybe for consistency we should keep doing that

Reply:
That's the default transformation that the latest version of IntelliJ applies when moving from prop to ctor injection.

It seems like good practice to use final as these properties should never be assigned outside the ctor - will remove the final keywords for now for the sake of consistency.